modules/hostap/src/supp_api.c

```diff
@@ -1408,6 +1410,75 @@ int hapd_config_network(struct hostapd_iface *iface,
 }
 #endif
 
+int supplicant_ap_status(const struct device *dev, struct wifi_iface_status *status)
+{
+	int ret = 0;
+	struct hostapd_iface *iface = NULL;
+	struct hostapd_config *conf = NULL;
+	struct hostapd_data *hapd = NULL;
+	struct hostapd_bss_config *bss = NULL;
+
+	k_mutex_lock(&wpa_supplicant_mutex, K_FOREVER);
+	iface = get_hostapd_handle(dev);
+	if (!iface) {
+		ret = -1;
+		wpa_printf(MSG_ERROR, "Interface %s not found", dev->name);
+		goto out;
+	}
+	conf = iface->conf;
+	if (!conf) {
+		ret = -1;
+		wpa_printf(MSG_ERROR, "Conf %s not found", dev->name);
+		goto out;
+	}
+	bss = conf->bss[0];
+	if (!bss) {
+		ret = -1;
+		wpa_printf(MSG_ERROR, "Bss_conf %s not found", dev->name);
+		goto out;
+	}
+	hapd = iface->bss[0];
+	if (!hapd) {
+		ret = -1;
+		wpa_printf(MSG_ERROR, "Bss %s not found", dev->name);
+		goto out;
+	}
+
+	int proto;
+	int key_mgmt;
+
+	status->state = iface->state;
+	struct hostapd_ssid *ssid = &bss->ssid;
+
+	os_memcpy(status->bssid, hapd->own_addr, WIFI_MAC_ADDR_LEN);
+	status->iface_mode = WPAS_MODE_AP;
+	status->band = wpas_band_to_zephyr(wpas_freq_to_band(iface->freq));
+	key_mgmt = bss->wpa_key_mgmt;
+	proto = bss->wpa;
+	status->security = wpas_key_mgmt_to_zephyr(key_mgmt, proto);
+	status->mfp = bss->ieee80211w;
+	status->channel = conf->channel;
+	os_memcpy(status->ssid, ssid->ssid, ssid->ssid_len);
+
+	status->dtim_period = bss->dtim_period;
+	status->beacon_interval = conf->beacon_int;
+
+	struct hostapd_hw_modes *hw_mode = iface->current_mode;
```

Comment:
Variable introduction to start of the function please

Reply:
modified